### PR TITLE
graphiql v3 blog

### PR DESCRIPTION
**Pull request by dimaMachina**

### CODE REVIEW

File: website/pages/blog/graphiql-3/index.mdx [hunk 1]
```diff
@@ -0,0 +1,212 @@
+---
+title: GraphiQL 3 Is Here and GraphiQL 4 Is Coming
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:2]: I would change the title to something that also gives information about the content
maybe like "GraphiQL v3 is here - React 18, better codebase, balblabla, and you can try v4 alpha with Monaco editor!"

File: website/pages/blog/graphiql-3/index.mdx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+tags: [graphql, react, graphiql]
+authors: dimitri
+date: 2023-07-01
+description: GraphiQL 3 was released, here is what was changed and what will be in GraphiQL 4.
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:6]: same here, more information about the content

File: website/pages/blog/graphiql-3/index.mdx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+10 months ago
+[GraphiQL 2 was released](https://github.com/graphql/graphiql/releases/tag/graphiql%402.0.0) with a
+completely new design and new components library named
+[`@graphiql/react`](https://github.com/graphql/graphiql/tree/main/packages/graphiql-react).
+
+Unfortunately previously planned in the [roadmap](https://github.com/graphql/graphiql/issues/2328)
+most-waited features as integration [monaco-editor](https://github.com/microsoft/monaco-editor) and
+new docs explorer (aka Query Builder) were not introduced.
+
+After months and months, no progress was made, and I was called to join forces and help the
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:24]: I would make it a bit nicer maybe
Maybe say that you decided to join the team and push all the things on the roadmap

File: website/pages/blog/graphiql-3/index.mdx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  **About me:** I am Dimitri POSTOLOV, software engineer from The Guild. You may know me as a
+  maintainer of [GraphQL-ESLint](https://the-guild.dev/graphql/eslint/docs) and
+  [Nextra](https://nextra.site).
+</Callout>
+
+## How I Started Maintaining Library
+
+### Put Out the Lights on the Christmas Tree 🎄
+
+When I started to maintain GraphiQL the first thing first was ESLint improvements, because ESLint
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:37]: maybe add something like that you need good tools and order for good long term maintenance, which is what The Guild is known for

File: website/pages/blog/graphiql-3/index.mdx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+   `@graphiql/plugin-code-exporter` [#3229](https://github.com/graphql/graphiql/pull/3228) and
+   [PR](https://github.com/graphql/graphiql/pull/3229).
+
+1. Removed unnecessary `<div />` wrappers [#3235](https://github.com/graphql/graphiql/pull/3235).
+
+1. Simplified Next.js and Vite examples with `monaco-graphql`
+   [#3140](https://github.com/graphql/graphiql/pull/3140) and
+   [#3281](https://github.com/graphql/graphiql/pull/3281).
+
+1. And
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:106]: I would add here a number of over 150 PRs

File: website/pages/blog/graphiql-3/index.mdx [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+| GraphiQL 4 🔜 | Monaco-Editor / Radix / Headless UI |
+
+<p className="mt-3 text-center italic">Table of comparison versions of GraphiQL</p>
+
+With all the above changes I also included a tab reorders feature:
+
+![Reorder tabs feature](./reorder-tabs.gif)
+
+{/* prettier-ignore */}
+<p className="mt-3 text-center italic">Tab reorders made by
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:131]: I would move that to the previous sentence, something like 

In v3 I've also included the tab reorders feature made by [@mylesmmurphy](https://github.com/mylesmmurphy)
  [#3111](https://github.com/graphql/graphiql/pull/3111):

File: website/pages/blog/graphiql-3/index.mdx [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+Try GraphiQL 3 by installing it with your favourite package manager:
+
+```sh npm2yarn
+npm i graphql@3
+```
+
+## GraphiQL 4 🔜
+
+14 months ago the initial issue was created on the integration `monaco-editor` in GraphiQL.
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:143]: I think originally the talk about monaco was many years earlier even

File: website/pages/blog/graphiql-3/index.mdx [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+{/* prettier-ignore */}
+<Anchor
+  href="https://github.com/graphql/graphiql/issues/2326#issue-1208525253" className="mt-6 block"
+>![Initial comment on integration of monaco-editor in GraphiQL](./img_4.png)</Anchor>
+
+but no real progress in GraphiQL was made (except
+[`graphiql-prototype`](https://github.com/jonathanawesome/graphiql-prototype) by
+[@jonathanawesome](https://github.com/jonathanawesome)).
+
+<Callout type="warning">
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:154]: I would phrase it a bit different and less offensive
Maybe you can say that there was some initial work on Jon on a prototype outside of GraphiQL without integration

File: website/pages/blog/graphiql-3/index.mdx [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+I already started on such difficult integration, here my initial PR
+[#3234](https://github.com/graphql/graphiql/pull/3234), or you can try it in the following
+[preview URL](https://deploy-preview-3234--graphiql-test.netlify.app) (the alpha version will be
+available soon too).
+
+{/* prettier-ignore */}
+<Anchor href="https://deploy-preview-3234--graphiql-test.netlify.app" className="mt-6 block">![GraphiQL
+  4 Preview](./img_3.png)</Anchor>
+
```
Comment (Urigo) [inline website/pages/blog/graphiql-3/index.mdx:174]: here is the current high level roadmap we have for GraphiQL, please comment and let us know what you think is missing:

**Review by Urigo — CHANGES_REQUESTED** [review]

**Comment by dimaMachina** [comment]
closing due to obsolescence

**Pull request closed without merging**